src/encoded/types/workflow.py

```diff
@@ -135,6 +140,367 @@
+
+        if not workflow_run_model or not hasattr(workflow_run_model, 'source'):
+            return
+
+        workflow_run_model_obj = workflow_run_model.source.get('object',{})
+        input_files = workflow_run_model_obj.get('input_files', [])
+        output_files = workflow_run_model_obj.get('output_files', [])
+
+        step = {
+            "uuid" : last_workflow_run_uuid, # Gets moved to stepNode.meta.uuid on front-end
+            "name" : workflow_run_model_obj.get('display_title'), # Gets duplicated to stepNode.meta.name, and stepNode.name,
+            "meta" : {
+                "status" : workflow_run_model_obj.get('status'),
+                '@type'  : workflow_run_model_obj.get('@type'),
+                '@id'    : workflow_run_model_obj.get('@id'),
+                'date_created' : workflow_run_model_obj.get('date_created')
+            },
+            "analysis_step_types" : [], # Gets moved to stepNode.meta.analysis_step_types on front-end (we could put into meta here as well)
+            "inputs" : [],
+            "outputs" : []
+        }
+
+        # Fill 'Analysis Step Types' w/ workflow name; TODO: Add component analysis_steps.
+        workflow_uuid = get_unique_key_from_at_id(workflow_run_model_obj.get('workflow')) #workflow_run.properties.get('workflow')
+        workflow_model = None
+        if workflow_uuid:
+            workflow_model = get_model_by_uuid(workflow_uuid)
+            if workflow_model and hasattr(workflow_model, 'source'):
+                workflow_model_obj = workflow_model.source.get('object',{})
+                if workflow_model_obj.get('workflow_type'):
+                    step['analysis_step_types'].append(workflow_model_obj['workflow_type'])
+                    step['meta']['workflow'] = {
+                        '@id'               : workflow_model_obj.get('@id') or workflow_run_model_obj.get('workflow'),
+                        '@type'             : workflow_model_obj.get('@type'),
+                        'title'             : workflow_model_obj.get('title'),
+                        'description'       : workflow_model_obj.get('description'),
+                        'display_title'     : workflow_model_obj.get('display_title') or workflow_model_obj.get('title'),
+                        'accession'         : workflow_model_obj.get('accession'),
+                        'workflow_steps'    : workflow_model_obj.get('workflow_steps'),
+                        'uuid'              : workflow_uuid,
+                        'date_created'      : workflow_model_obj.get('date_created'),
+                        'link_id'           : workflow_model_obj.get('link_id'),
+                        'workflow_type'     : workflow_model_obj.get('workflow_type'),
+                        'cwl_pointer'       : workflow_model_obj.get('cwl_pointer')
+                    }
+
+
+        # Add Output Files, 1-level deep max (maybe change in future)
```

Comment:
this is basically same code as for input files... probably refactor into resuable function and then it should be easier to write some tests

Reply:
Will do. Likely will wait until get to writing trace_future function. Reasoning is that when tracing history, we also trace input files (see step['inputs']['source'] -- calls recursive tracing function -- vs step['outputs']['target'] -- where just have list with 1 item until maybe a subsequently traced input files 'claims/matches' it and adds to it in lines 460-472). I haven't written `generate_targets_for_output` yet (or converted `generate_sources_for_input` to work for both).